### PR TITLE
style(app): overflow btn and navbar styling

### LINKED ISSUE #10628: 6.0 Feedback:  Issue with component colours
### Overview

Noticed that in two instances, the colours that we're using are off. 
- Navbar highlight state
- Menu overflow states

### Current Behavior

[Link to view annotated frame in Figma](https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=13903%3A276554)

**Navbar**
- [x] (1) Update colour of the hover state to #24313F. This is our `darkBlackHover`
- [x] (2) Update the focus state. It's our hover state background with our focus colour as a stroke. 

Screen Shot 2022-06-06 at 2 10 44 PM

**menu overflow icon**
Screen Shot 2022-06-06 at 2 15 13 PM

*We need to separate the focus state from the selected/active state. The focus state should only appear when the user is manually tabbing through the links using a keyboard. This is an accessibility-specific state and should not appear otherwise. 

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Operating system

_No response_

### Robot setup or anything else?

_No response_

### PR DESCRIPTION
**Pull request by jerader**
closes #10628 

# Overview

Overflow menu button previously had styling and states that didn't match designs. This fixes that along with correcting the `darkBlackHover` color in the constants and fixing the focus state with the navbar.

# Changelog

- change `overflowBtn`, `Navbar`, and the `darkBlackHover` color in the constants.

# Review requests

- look at the overflow menu buttons, do the states match designs?
- look at the navbar menu items, do the states match designs?

# Risk assessment

low

### CODE REVIEW

File: app/src/atoms/MenuList/OverflowBtn.tsx [hunk 1]
```diff
@@ -7,11 +7,11 @@ const overflowButtonStyles = css`
...
 
   &:hover {
-    background-color: ${COLORS.darkGreyHover};
+    background-color: #16212d15;
   }
 
   &:active {
-    background-color: ${COLORS.darkBlack};
+    background-color: #16212d25;
     box-shadow: 0 0 0 1px ${COLORS.darkBlack};
```
Comment (koji) [inline app/src/atoms/MenuList/OverflowBtn.tsx]: I think we don't need box-shadow for active.

Screen Shot 2022-06-09 at 10 16 24 AM

File: app/src/atoms/MenuList/__tests__/OverflowBtn.test.tsx [hunk 2]
```diff
@@ -41,7 +41,7 @@ describe('OverflowBtn', () => {
 
     expect(getByRole('button')).toHaveStyleRule(
       'background-color',
-      COLORS.darkBlack,
+      `#16212d25`,
```
Comment (koji) [inline app/src/atoms/MenuList/__tests__/OverflowBtn.test.tsx]: probably `#16212d40` would be better.

Screen Shot 2022-06-09 at 10 59 31 AM


Also, I tried this.
https://rgbacolorpicker.com/rgba-to-hex

File: app/src/atoms/MenuList/__tests__/OverflowBtn.test.tsx [hunk 3]
```diff
@@ -27,7 +27,7 @@ describe('OverflowBtn', () => {
 
     expect(getByRole('button')).toHaveStyleRule(
       'background-color',
-      COLORS.darkGreyHover,
+      `#16212d15`,
```
Comment (koji) [inline app/src/atoms/MenuList/__tests__/OverflowBtn.test.tsx]: Probably `#16212d26`

Comment (koji) [reply app/src/atoms/MenuList/__tests__/OverflowBtn.test.tsx]: Also, once #10664 gets approval from devs, I will update this to `COLORS.colorName`.

**Review by koji — CHANGES_REQUESTED** [review]
Attached the current `edge`
https://user-images.githubusercontent.com/474225/172880182-915b5508-6454-42b0-939c-a3ecf627b41f.mov

Could you check one thing about `focus`
As you can see, when we click one of the nav menu items, the app doesn't show focus box-shadow.
On the other hand, this branch shows focus box-shadow when clicking one of the nav menu items.

File: app/src/atoms/MenuList/__tests__/OverflowBtn.test.tsx (hunk 3 above)
Comment (koji) [reply app/src/atoms/MenuList/__tests__/OverflowBtn.test.tsx]: This gist might be useful to know how to convert opacity into hex.
https://gist.github.com/lopspower/03fb1cc0ac9f32ef38f4

Comment (jerader) [reply app/src/atoms/MenuList/__tests__/OverflowBtn.test.tsx]: thanks for that! I forgot that the numbers dont translate 1:1

**Comment by mmencarelli** [comment]
[Figma annotation space](https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=14136%3A276576)

[Loom video explaining feedback for nav](https://www.loom.com/share/8c1666055bc442289c7a3f9108da5fdb)

- Need to update the hover state to our darkGreyHover (#24313F)
- Need to de-couple pressed and focus state. Focus state (orange outline) only appears when the user is tabbing through the links 

Feedback for overflow buttons
- Need to update all the states
- Need to decouple focus state from pressed and active state
- 
Screen Shot 2022-06-09 at 1 54 22 PM

**Comment by mmencarelli** [comment]
looking good!!!!

**Review by mmencarelli — APPROVED** [review]

File: app/src/atoms/MenuList/OverflowBtn.tsx [hunk 4]
```diff
@@ -7,30 +7,32 @@ const overflowButtonStyles = css`
...
-    fill: ${COLORS.darkBlack};
+  &:active circle,
+  &:focus circle {
+    fill: ${COLORS.darkGreyPressed};
   }
 
-  &:focus {
-    box-shadow: 0 0 0 3px ${COLORS.darkGreyPressed};
+  &:focus-visible {
+    box-shadow: 0 0 0 3px ${COLORS.warning};
```
Comment (koji) [inline app/src/atoms/MenuList/OverflowBtn.tsx:27]: I think we need to add `background-color: ${COLORS.transparent};`
Just sent a message to ask about this to Mel.

https://www.figma.com/file/l7BAJAGICr10ELsSqQD0Sr?node-id=2180:85399#209555192

Comment (jerader) [reply app/src/atoms/MenuList/OverflowBtn.tsx:27]: shouldn't it already be transparent since we didn't specify a `background-color` for the default state?

**Review by koji — APPROVED** [review]
Thank you for fixing the focus state thing! ⭐️
Leave one comment on overflow menu focus, but other changes look good to me!

File: app/src/atoms/MenuList/OverflowBtn.tsx (hunk 4 above)
Comment (koji) [reply app/src/atoms/MenuList/OverflowBtn.tsx:27]: Yeah, you are right. We should specify `background-color`.
maybe `COLORS.white` or `COLORS.background` ?

Comment (jerader) [reply app/src/atoms/MenuList/OverflowBtn.tsx:27]: whoops i merged before seeing this reply, sorry. I would want to confirm with Mel what we would want the background color to be - in every usage of the `overflowBtn`, it seems it is being used above a white background but if we add it to something without a white background in the future, maybe we would want the background color to be transparent.

**Pull request merged**